src/twisted/internet/test/test_endpoints.py

```diff
@@ -3259,6 +3259,29 @@ def test_sslNoTrailingNewlinePem(self):
         ctx = server._sslContextFactory.getContext()
         self.assertIsInstance(ctx, ContextType)
 
+    @skipIf(skipSSL, skipSSLReason)
+    def test_sslWithCustomCipher(self):
+        """
+        Test adding custom cipher to the serverFromString function.
+        """
+        reactor = object()
+        server = (
+            endpoints.serverFromString(
+                reactor,
+                "ssl:1234:backlog=12:privateKey=%s:"
+                "certKey=%s:sslmethod=TLSv1_2_METHOD:interface=10.0.0.1:cipher=ECDHE-ECDSA-AES128-GCM-SHA256,ECDHE-ECDSA-CHACHA20-POLY1305,ECDHE-RSA-AES128-GCM-SHA256"
```

Comment:
WOuld it make sense to keep the input for this test with a narrow scope?

Do we need backlog here ?
Do we need to care about sslmethod here ? 

I expect that we have separate tests for backlog or ssl method

```suggestion
                "certKey=%s:cipher=ECDHE-ECDSA-AES128-GCM-SHA256,ECDHE-RSA-AES128-GCM-SHA256"
```

Reply:
Maybe it would help to have a cipher text like

```
ALL:!ADH:@STRENGTH:+RSA:-DSA
```

I hope that this provide a good cover of the value that can be found in a cipher definition